### PR TITLE
  support git shortlog from upstream git for creating commit messages

### PR DESCRIPTION
**Pull request by fenrus75**

### CODE REVIEW

File: autospec/commitmessage.py [hunk 1]
```diff
@@ -168,6 +170,43 @@ def process_NEWS(newsfile):
...
+        return ""
+    if oldversion == newversion:
+        return ""
+
+    run(["git", "-C", "results", "clone", giturl])
+    p = run(["git", "-C", "results/" + tarball.name, "tag"], stdout=PIPE)
+    tags = p.stdout.decode('utf-8').split('\n')
+
+    for t in tags:
+        i = t.find(oldversion)
```
Comment (bryteise) [inline autospec/commitmessage.py]: This seems like it could introduce a lot of noise for projects that do version as 1, 2, etc for their versions.

Comment (fenrus75) [reply autospec/commitmessage.py]: is there a "ends-width" thing in python then?

**Comment by bryteise** [comment]
"string-suffix".endswith("suffix") is what you want.

**Comment by matthewrsj** [comment]
+1

**Pull request merged**